File: app/src/main/java/fr/paug/androidmakers/model/Speaker.java
package fr.paug.androidmakers.model;

import android.support.annotation.Nullable;
import android.text.TextUtils;

import java.util.List;

/**
 * Created by stan on 18/03/2017.
 */
public class Speaker {

    public final String name;
    public final String bio;
    public final String company;
    public final String surname;
    public final String thumbnailUrl;
    public final List<Ribbon> ribbonList;
    public final boolean rockstar;
    public final List<SocialNetworkHandle> socialNetworkHandles;

    public Speaker(String name, String bio, String company, String surname, String thumbnailUrl, String rockstar, List<SocialNetworkHandle> socialNetworkHandleList, List<Ribbon> ribbonList) {
        this.name = name;
        this.bio = bio;
        this.company = company;
        this.surname = surname;
        this.thumbnailUrl = thumbnailUrl;
        this.rockstar = Boolean.parseBoolean(rockstar);
        this.socialNetworkHandles = socialNetworkHandleList;
        this.ribbonList = ribbonList;
    }

    public String getFullNameAndCompany() {
        return this.name + " " + this.surname + (TextUtils.isEmpty(company) ? "" : ", " + this.company);
    }

    /**
     * Gets the main ribbon.
     * The main ribbon is the first ribbon of the list.
     *
     * @return the main ribbon, or null if the ribbon list is empty or null.
     */
    @Nullable
    public Ribbon getMainRibbon() {
        if (ribbonList != null && !ribbonList.isEmpty()) {
            return ribbonList.get(0);
        }
        return null;
    }

}

File: app/src/main/java/fr/paug/androidmakers/model/Speaker.kt
package fr.paug.androidmakers.model

import android.text.TextUtils
import androidx.annotation.StringRes
import fr.paug.androidmakers.R
import fr.paug.androidmakers.model.SocialNetworkHandle
import fr.paug.androidmakers.model.Ribbon
import androidx.annotation.DrawableRes
import fr.paug.androidmakers.model.SocialNetworkHandle.SocialNetworkType

class Speaker(
    val name: String,
    val bio: String,
    val company: String,
    val surname: String,
    val thumbnailUrl: String,
    rockstar: String?,
    socialNetworkHandleList: List<SocialNetworkHandle>,
    ribbonList: List<Ribbon>?
) {
    val ribbonList: List<Ribbon>?
    val rockstar: Boolean
    val socialNetworkHandles: List<SocialNetworkHandle>
    val fullNameAndCompany: String
        get() = name + " " + surname + if (TextUtils.isEmpty(company)) "" else ", " + company
    val fullName: String
        get() = name + " " + surname

    /**
     * Gets the main ribbon.
     * The main ribbon is the first ribbon of the list.
     *
     * @return the main ribbon, or null if the ribbon list is empty or null.
     */
    val mainRibbon: Ribbon?
        get() = if (ribbonList != null && !ribbonList.isEmpty()) {
            ribbonList[0]
        } else null

    init {
        this.rockstar = java.lang.Boolean.parseBoolean(rockstar)
        socialNetworkHandles = socialNetworkHandleList
        this.ribbonList = ribbonList
    }
}